server/src/main/java/org/apache/cassandra/sidecar/routes/cdc/ListCDCDirHandler.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.cassandra.sidecar.routes.AbstractHandler;
+import org.apache.cassandra.sidecar.utils.CDCUtil;
+import org.apache.cassandra.sidecar.utils.CassandraInputValidator;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+
+import static org.apache.cassandra.sidecar.utils.CDCUtil.getIdxFileName;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.getLogFilePrefix;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.isIndexFile;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.parseIndexFile;
+
+/**
+ * Provides REST endpoint for listing commit logs in CDC directory.
+ */
+@Singleton
+public class ListCDCDirHandler extends AbstractHandler<Void>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ListCDCDirHandler.class);
+    private final ServiceConfiguration config;
+
+    @Inject
+    public ListCDCDirHandler(InstanceMetadataFetcher metadataFetcher,
+                             SidecarConfiguration config,
+                             ExecutorPools executorPools,
+                             CassandraInputValidator validator)
+    {
+        super(metadataFetcher, executorPools, validator);
+        this.config = config.serviceConfiguration();
+    }
+
+    @Override
+    protected void handleInternal(RoutingContext context,
+                                  HttpServerRequest httpRequest,
+                                  String host,
+                                  SocketAddress remoteAddress,
+                                  Void request)
+    {
+        String cdcDir = metadataFetcher.instance(host).cdcDir();
+        ListCDCSegmentResponse listCDCSegmentResponse = new ListCDCSegmentResponse(config.host(),
+                                                                                   config.port(),
+                                                                                   new ArrayList<>());
+        try
+        {
+            addResources(cdcDir, listCDCSegmentResponse);
+        }
+        catch (IOException e)
+        {
+            LOGGER.warn("Error listing the CDC commit log segments", e);
+            context.response()
+                   .setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code())
+                   .setStatusMessage(e.getMessage())
+                   .end();
+            return;
+        }
+
+        context.response().end(Json.encodePrettily(listCDCSegmentResponse));
+    }
+
+    private void addResources(String cdcDirPath, ListCDCSegmentResponse listResponse) throws IOException
+    {
+        File cdcDir = Paths.get(cdcDirPath).toAbsolutePath().toFile();
+        if (!cdcDir.isDirectory())
+        {
+            throw new IOException("CDC directory does not exist");
+        }
+
+        File[] cdcFiles = cdcDir.listFiles();
+        if (cdcFiles == null || cdcFiles.length == 0)
+        {
+            return;
+        }
+
+        Set<String> idxFileNamePrefixes = Arrays.stream(cdcFiles)
```

Comment:
Maybe a nitpick, but in the main Cassandra project we decided to avoid streams on hot paths. 
https://lists.apache.org/thread/65glsjzkmpktzmns6j9wvr4nczvskx36

Maybe the performance issue is not as critical in this component, but it'd be nice to keep it aligned.